include/cereal/archives/json.hpp

```diff
@@ -286,7 +286,7 @@ namespace cereal
       void saveLong(T lu){ saveValue( static_cast<std::uint64_t>( lu ) ); }
 
     public:
-#ifdef _MSC_VER
+#if defined(_MSC_VER) && !defined(__clang__)
```

Comment:
I'm not sure you'd want to do this, clang-cl has the same integer sizes as MSVC, so we'd want the unsigned long overload as well here, right?

Reply:
to suppress this error

```
call to member function 'saveValue' is ambiguous
```

in json.hpp

Reply:
Ok correction: in our org we never came across this even though our `json.hpp` looks the same. We're using clang-cl version 8.0.0. I wonder if @durswd has a different configuration whereby `sizeof(long)` is coming out as 8?